src/main/scala/scalaz/ziokeeper/VectorClock.scala

```diff
@@ -0,0 +1,64 @@
+package scalaz.ziokeeper
+
+import scala.annotation.tailrec
+
+case class VectorClock[A](value: Map[A, Long])
+
+object VectorClock {
+  def empty[A] = VectorClock[A](value = Map.empty)
+
+  implicit class VectorClockOps[A](val v: VectorClock[A]) extends AnyVal {
+    def time(mid: A): Long = v.value.getOrElse(mid, 0L)
+
+    def increment(mid: A, value: Long): VectorClock[A] =
+      VectorClock(v.value.updated(mid, time(mid) + value))
+  }
+
+  sealed trait CompareVectorClock
+
+  case object TheSame extends CompareVectorClock
```

Comment:
What's the difference between `TheSame` and `Concurrent`?

Reply:
the idea behind vector clock is pretty simple 
- increment you local counter whatever you perform actions(e.g. add, remove, modify) as well as when you receive messages from other members of the cluster.

when we have cluster with three member with ids 111, 222, 333 vector clock initially will look like this on all nodes.
(111 -> 0, 222 -> 0, 333 -> 0) 

at this point the are all **TheSame**

imagine that 111 and 222 perform add action 
111 = (111 -> 1, 222 -> 0, 333 -> 0)
222 = (111 -> 0, 222 -> 1, 333 -> 0)
333 =  (111 -> 0, 222 -> 0, 333 -> 0)
111 and 222 are concurrent because you cannot say which happen first.

let me know if that clear enough.